config.h.in

```diff
@@ -1,329 +1,16 @@
-
-/* Define if you have the `sysctlbyname' function. */
-#cmakedefine HAVE_SYSCTLBYNAME
-
-/* Define if you have the `syssgi' function. */
-#cmakedefine HAVE_SYSSGI
-
-/* Define if you have the <sys/mman.h> header file. */
-#cmakedefine HAVE_SYS_MMAN_H
-
-/* Define if you have the <sys/param.h> header file. */
-#cmakedefine HAVE_SYS_PARAM_H
-
-/* Define if you have the <sys/processor.h> header file. */
-#cmakedefine HAVE_SYS_PROCESSOR_H
-
-/* Define if you have the <sys/resource.h> header file. */
-#cmakedefine HAVE_SYS_RESOURCE_H
-
-/* Define if you have the <sys/stat.h> header file. */
-#cmakedefine HAVE_SYS_STAT_H		1
-
-/* Define if you have the <sys/sysctl.h> header file. */
-#cmakedefine HAVE_SYS_SYSCTL_H
-
-/* Define if you have the <sys/syssgi.h> header file. */
-#cmakedefine HAVE_SYS_SYSSGI_H
-
-/* Define if you have the <sys/systemcfg.h> header file. */
-#cmakedefine HAVE_SYS_SYSTEMCFG_H
-
-/* Define if you have the <sys/times.h> header file. */
-#cmakedefine HAVE_SYS_TIMES_H
-
-/* Define if you have the <sys/time.h> header file. */
-#cmakedefine HAVE_SYS_TIME_H
-
-/* Define if you have the <sys/types.h> header file. */
-#cmakedefine HAVE_SYS_TYPES_H	1
-
-/* Define if you have the `times' function. */
-#cmakedefine HAVE_TIMES
-
-/* Define if you have the <unistd.h> header file. */
-#cmakedefine HAVE_UNISTD_H
-
-/* Define if you have vsnprintf and it works properly. */
-#cmakedefine HAVE_VSNPRINTF
-
-/* Assembler local label prefix */
-#cmakedefine LSYM_PREFIX
-
-/* Define if you have the `fesetround' function via the <fenv.h> header file.
-   */
-#cmakedefine MPFR_HAVE_FESETROUND
-
-#cmakedefine HAVE_SSTREAM 1
-
-/* Define if compiler has function prototypes */
-#cmakedefine PROTOTYPES			1
-
-/* Define as the return type of signal handlers (`int' or `void'). */
-#cmakedefine RETSIGTYPE	void
-
-/* The size of a `unsigned long', as computed by sizeof. */
-#cmakedefine SIZEOF_UNSIGNED_LONG	4
-
-/* Define if sscanf requires writable inputs */
-#cmakedefine SSCANF_WRITABLE_INPUT
-
-/* Define if you have the ANSI C header files. */
-#cmakedefine STDC_HEADERS		1
-
 /* ./configure --enable-assert option, to enable some ASSERT()s */
-#cmakedefine WANT_ASSERT
+#cmakedefine FLINT_WANT_ASSERT
 
 /* Define if you cpu_set_t in sched.h */
-#cmakedefine01 HAVE_CPU_SET_T
-
-/* Define if your processor stores words with the most significant byte first
-   (like Motorola and SPARC, unlike Intel and VAX). */
-#cmakedefine WORDS_BIGENDIAN
-
-#cmakedefine HAVE_PTHREAD 1
+#cmakedefine01 FLINT_USES_CPUSET
 
-/* Define as `__inline' if that's what the C compiler calls it, or to nothing
-   if it is not supported. */
-#ifndef __cplusplus
-#define inline	__inline
-#endif
-
-#ifdef HAVE_STDINT_H
-#define HAVE_INTMAX_T        1
-#define HAVE_UINTMAX_T       1
-#define HAVE_PTRDIFF_T       1
-#define HAVE_UINT_LEAST32_T  1
-#define SIZEOF_UINTMAX_T	 8
-#endif
-
-#define NPRINTF_J            1
-#define NPRINTF_T            1
-
-#ifdef _MSC_VER
```

Comment:
the removal of this `#ifdef` section looks dodgy. E.g. definition of `alloca` is certainly needed. Perhaps cmake has a better way to handle this, @isuruf might know, as I imagine it's a common portability issue.

Also, perhaps `_alloca` should be `_malloca`, as https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/alloca?view=vs-2019 seems to recommend - but that's another story.

Reply:
Restored that part.